src/snarky.d.ts

```diff
@@ -51,489 +53,767 @@ declare interface ProvablePure<T> extends Provable<T> {
+   * ```
+   * 
+   * **Important: All the functions defined on a Field (arithmetic, logic, etc.) takes their arguments as `field-like`. A {@link Field} itself is also defined as a field-like element.
+   * 
+   * @param value - the value to coerce to a {@link Field}
+   * 
+   * @return A {@link Field} element which the value coerced from the argument in the pasta field.
    */
-  constructor(x: Field | number | string | boolean | bigint);
+  constructor(value: Field | number | string | boolean | bigint);
 
   /**
-   * Negates this {@link Field}. This is equivalent to multiplying the {@link Field}
-   * by -1.
+   * Negate a {@link Field}. This is equivalent to multiplying the {@link Field} by -1.
    *
-   * ```typescript
+   * ```ts
    * const negOne = Field(1).neg();
    * negOne.assertEquals(-1);
    * ```
+   * 
+   * ```ts
+   * const someField = Field(42);
+   * const negation = someField.neg();
+   * 
+   * negation.neg().assertEquals(someField.mul(Field(-1))); // This statement is always true regardless of the value of `someField`
+   * ```
+   * 
+   * **Warning: This is a modular negation. Please see `sub()` function for more details.
+   * 
+   * @return A {@link Field} element that is equivalent to the element multiplied by -1.
    */
   neg(): Field;
 
   /**
-   * Inverts this {@link Field} element.
+   * Invert this {@link Field} element. This is equivalent to equalize the {@link Field} to 1 over its value.
    *
    * ```typescript
-   * const invX = x.inv();
-   * invX.assertEquals(Field(1).div(x));
+   * const someField = Field(42);
+   * const inverse = someField.inv();
+   * inverse.assertEquals(Field(1).div(example)); // This statement is always true regardless of the value of `someField`
    * ```
+   * 
+   * **Warning: This is a modular inverse. Please see `div()` function for more details.
    *
    * @return A {@link Field} element that is equivalent to one divided by this element.
    */
   inv(): Field;
 
   /**
-   * Adds this {@link Field} element to another to a {@link Field} element.
+   * Add a `field-like` element to a {@link Field} element and returns the result.
+   *
+   * ```ts
+   * const value1 = Field(3);
+   * const value2 = 5; // A `field-like` element
+   * 
+   * const sum = value1.add(value2);
+   * 
+   * sum.assertEquals(Field(8));
+   * ```
+   * 
+   * **Warning: This is a modular addition in the pasta field.
    *
    * ```ts
-   * let a = Field(3);
-   * let sum = a.add(5)
+   * const value1 = Field(1);
+   * const value2 = Field(-7);
+   * 
+   * const sum = value1.add(value2);
+   * 
+   * // If you try to print sum - `console.log(sum.toBigInt())` - you will realize that it prints a very big integer, so you cannot use arithmetic operations on snarkyJS as in regular programming languages.
```

Comment:
```suggestion
   * // If you print sum - `console.log(sum.toBigInt())` - you will see a very big integer, because this is modular arithmetic and 1 + (-7) circles around the field to become p - 6
```
I think its wrong to say that "you cannot use arithmetic operations on snarkyJS as in regular programming languages", for several reasons:
* snarkyjs isn't a programming language. JavaScript is the programming language.
* you can use Field arithmetic, and it works fine, it's just different
* there is no "default arithmetic in regular programming languages". In many languages, you have int32 and int64 arithmetic which is modulo ranges like [-2^31, 2^31 - 1]. Then you have floats which are different again, etc

Reply:
@yunus433 can you change the wording across all comments?

I'm curious what others think about this though, @MartinMinkov @bkase